rust/src/treefile.rs

```diff
@@ -713,8 +708,12 @@ remove-files:
         let buf = buf.as_bytes();
         let mut input = io::BufReader::new(buf);
         match treefile_parse_stream(InputFormat::YAML, &mut input, Some(ARCH_X86_64)) {
-            Err(ref e) if e.kind() == io::ErrorKind::InvalidInput => {}
-            Err(ref e) => panic!("Expected invalid treefile, not {}", e.to_string()),
+            Err(ref e) => {
+                match e.downcast_ref::<io::Error>() {
```

Comment:
Interesting.  I need to learn more about the `TypeId` stuff, I guess it boils down to a unique id for a vtable or so?